drivers/include/sdcard_spi.h

```diff
@@ -0,0 +1,359 @@
+/*
+ * Copyright (C) 2016
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers
+ * @brief       Low-level driver for reading and writing sd-cards via spi interface.
+ * @{
+ *
+ * @file
+ * @brief       Low-level driver for reading and writing sd-cards via spi interface.
+ *              For details of the sd card standard and the spi mode refer to
+ *              "SD Specifications Part 1 Physical Layer Simplified Specification".
+ *              References to the sd specs in this file apply to Version 5.00 from August 10, 2016.
+ *              See https://www.sdcard.org/downloads/pls/pdf/part1_500.pdf for further details.
+ *
+ * @author      Michel Rottleuthner <michel.rottleuthner@haw-hamburg.de>
+ */
+
+#ifndef SDCARD_SPI_H
+#define SDCARD_SPI_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "periph/spi.h"
+#include "periph/gpio.h"
+
+
+/* number of clocks that should be applied to the card on init
+   before taking furter actions (see sd spec. 6.4.1.1 Power Up Time of Card) */
+#define SD_POWERSEQUENCE_CLOCK_COUNT 74
+
+/* R1 response bits (see sd spec. 7.3.2.1 Format R1) */
+#define SD_R1_RESPONSE_PARAM_ERROR       0b01000000
+#define SD_R1_RESPONSE_ADDR_ERROR        0b00100000
+#define SD_R1_RESPONSE_ERASE_SEQ_ERROR   0b00010000
+#define SD_R1_RESPONSE_CMD_CRC_ERROR     0b00001000
+#define SD_R1_RESPONSE_ILLEGAL_CMD_ERROR 0b00000100
+#define SD_R1_RESPONSE_ERASE_RESET       0b00000010
+#define SD_R1_RESPONSE_IN_IDLE_STATE     0b00000001
+#define SD_INVALID_R1_RESPONSE           0b10000000
+
+#define IS_VALID_R1_BYTE(X)   (((X) >> 7) == 0)
+#define R1_HAS_PARAM_ERR(X)   ((((X) &SD_R1_RESPONSE_PARAM_ERROR) != 0))
+#define R1_HAS_ADDR_ERR(X)    ((((X) &SD_R1_RESPONSE_ADDR_ERROR) != 0))
+#define R1_HAS_ERASE_ERR(X)   ((((X) &SD_R1_RESPONSE_ERASE_SEQ_ERROR) != 0))
+#define R1_HAS_CMD_CRC_ERR(X) ((((X) &SD_R1_RESPONSE_CMD_CRC_ERROR) != 0))
+#define R1_HAS_ILL_CMD_ERR(X) ((((X) &SD_R1_RESPONSE_ILLEGAL_CMD_ERROR) != 0))
+#define IS_R1_IDLE_BIT_SET(X) (((X) &SD_R1_RESPONSE_IN_IDLE_STATE) != 0)
+#define R1_HAS_ERROR(X) (R1_HAS_PARAM_ERR(X) || R1_HAS_ADDR_ERR(X) || R1_HAS_ERASE_ERR(X) || R1_HAS_CMD_CRC_ERR(X) || R1_HAS_ILL_CMD_ERR(X))
```

Comment:
Please keep the line length to max [80 characters](https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions#general) (100 if it absolutely doesn't go any other way)